### PR TITLE
drop capability to specify image in install-config

### PR DESCRIPTION
**Pull request by abhinavdahiya**
includes commit from https://github.com/openshift/installer/pull/812

RHCOS image that needs to be used for installation must be sources from RHCOS build pipeline and RHCOS build.
Keeping the image related fields in install-config allows users to change these values as part of valid configuration, but we do not want users to configure this option as
the RHCOS image controls the runtime and kubelet versions we depend on.

added `OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE` env with warning to override the image that will be used.

/cc @crawford @wking

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
> Changes look good.
> 
> Can we remove this function?
> 
> [installer/pkg/types/openstack/validation/realvalidvaluesfetcher.go](https://github.com/openshift/installer/blob/bf0c0de4d5158150095dd053c669b819bdb076b2/pkg/types/openstack/validation/realvalidvaluesfetcher.go#L66)
> 
> Line 66 in [bf0c0de](/openshift/installer/commit/bf0c0de4d5158150095dd053c669b819bdb076b2)
>  func (f realValidValuesFetcher) GetImageNames(cloud string) ([]string, error) {

done with https://github.com/openshift/installer/pull/1052/commits/83d9bf8d523e4d8664304d710e3eb06e287084dc

File: pkg/asset/cluster/tfvars.go [hunk 1]
```diff
@@ -34,6 +35,7 @@ func (t *TerraformVariables) Name() string {
 func (t *TerraformVariables) Dependencies() []asset.Asset {
 	return []asset.Asset{
 		&installconfig.InstallConfig{},
+		new(rhcos.Image),
```
Comment (wking) [inline pkg/asset/cluster/tfvars.go]: Why is this not `&rhcos.Image{}` like its neighbors?

Comment (abhinavdahiya) [reply pkg/asset/cluster/tfvars.go]: https://play.golang.org/p/NUhwmraqAtM
```console
prog.go:10:7: invalid pointer type *Image for composite literal
```

Comment (abhinavdahiya) [reply pkg/asset/cluster/tfvars.go]: https://golang.org/ref/spec#Address_operators

File: pkg/asset/rhcos/image.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	var err error
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer cancel()
+	switch config.Platform.Name() {
+	case aws.Name:
+		osimage, err = rhcos.AMI(ctx, rhcos.DefaultChannel, config.Platform.AWS.Region)
+	case libvirt.Name:
+		osimage, err = rhcos.QEMU(ctx, rhcos.DefaultChannel)
+	case openstack.Name:
+		osimage = "rhcos"
```
Comment (wking) [inline pkg/asset/rhcos/image.go]: This is much-simplified from the previous `GetImageNames` you're removing in 83d9bf8d523.  @hardys added that image-name select widget in #766, and sometimes OpenStack doesn't have reliable defaults (e.g. #788).  Are we comfortable with a hard-coded value here, or do we need to retain some lookup logic or user input?

**Comment by abhinavdahiya** [comment]
> Drop:
> 
> > added `OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE` env with warning to override the image that will be used.
> 
> from your commit message (and maybe add it to [8957997](https://github.com/openshift/installer/commit/89579974bca7cc94ac05f753bd3a6f617cc871ea), since that's the commit adding the variable)?

Yeah, my original thought was somebody seeing this commit of dropping image fields from InstallConfig might benefit from it... but i can move it if you like?

**Comment by wking** [comment]
> Yeah, my original thought was somebody seeing this commit of dropping image fields from InstallConfig might benefit from it...

Maybe mention "in follow-up work" in the commit message?  Or squash this PR down a bit, currently there are commits removing properties from structs before commits removing the property consumers and things like that.  On the other hand, it's nice to see smaller pivots too.  I don't really care either way at this stage ;).

**Comment by wking** [comment]
[e2e-aws][1] had lots of:

```
fail [github.com/openshift/origin/test/extended/builds/build_pruning.go:52]: Expected error:
    <*errors.errorString | 0xc4217d9ac0>: {
        s: "Failed to import expected imagestreams",
    }
    Failed to import expected imagestreams
not to have occurred
```

ending up with:

```
Failing tests:

[Conformance][Area:Networking][Feature:Router] The HAProxy router should expose prometheus metrics for a route [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds] forcePull should affect pulling builder images  ForcePull test case execution custom [Suite:openshift/conformance/parallel]
[Feature:Builds] forcePull should affect pulling builder images  ForcePull test case execution docker [Suite:openshift/conformance/parallel]
[Feature:Builds] forcePull should affect pulling builder images  ForcePull test case execution s2i [Suite:openshift/conformance/parallel]
[Feature:Builds][Conformance] oc new-app  should fail with a --name longer than 58 characters [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance] oc new-app  should succeed with a --name of 58 characters [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should fail resolving unresolvable valueFrom in docker build environment variable references [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should fail resolving unresolvable valueFrom in sti build environment variable references [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should successfully resolve valueFrom in docker build environment variables [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance][valueFrom] process valueFrom in build strategy environment variables  should successfully resolve valueFrom in s2i build environment variables [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  [Conformance] buildconfigs should have a default history limit set when created via the group api [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  [Conformance] buildconfigs should not have a default history limit set when created via the legacy api [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune builds after a buildConfig change [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune canceled builds based on the failedBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune completed builds based on the successfulBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune errored builds based on the failedBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune failed builds based on the failedBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:DeploymentConfig] deploymentconfigs when run iteratively [Conformance] should immediately start a new deployment [Suite:openshift/conformance/parallel/minimal]
[Feature:DeploymentConfig] deploymentconfigs with multiple image change triggers [Conformance] should run a successful deployment with a trigger used by different containers [Suite:openshift/conformance/parallel/minimal]
[Feature:DeploymentConfig] deploymentconfigs with multiple image change triggers [Conformance] should run a successful deployment with multiple triggers [Suite:openshift/conformance/parallel/minimal]
[image_ecosystem][mongodb] openshift mongodb image  creating from a template should instantiate the template [Suite:openshift/conformance/parallel]
```

/retest

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1052/pull-ci-openshift-installer-master-e2e-aws/2834/build-log.txt

**Comment by dgoodwin** [comment]
Breaking change for Hive. :( But the env var should cover us for now. Will raise in arch call about how we should handle this image ID long term, across upgrades, in an external system like Hive.

**Pull request merged**